apps/dash-oil-and-gas/app.py

```diff
@@ -450,14 +453,17 @@ def make_main_figure(
         )
         traces.append(trace)
 
-    if main_graph_layout is not None and "locked" in selector:
-
-        lon = float(main_graph_layout["mapbox"]["center"]["lon"])
-        lat = float(main_graph_layout["mapbox"]["center"]["lat"])
-        zoom = float(main_graph_layout["mapbox"]["zoom"])
-        layout["mapbox"]["center"]["lon"] = lon
-        layout["mapbox"]["center"]["lat"] = lat
-        layout["mapbox"]["zoom"] = zoom
+    if main_graph_layout is not None and selector is not None and "locked" in selector:
+        if (
+            "mapbox.center" in main_graph_layout.keys()
+        ):  # relayoutData is None by default, and {'autosize': True} without relayout action
```

Comment:
@ycaokris I think this might be a tiny bit easier to read if you format the comment above the if statement and make that 1 line:  
```
# relayoutData is None by default, and {'autosize': True} without relayout action
if "mapbox.center" in main_graph_layout.keys():
```
or incorporate it into the above if statement